control-plane/pkg/reconciler/trigger/trigger.go

```diff
@@ -326,11 +347,61 @@ func isKnativeKafkaBroker(broker *eventing.Broker) (bool, string) {
 
 func deliveryOrderFromString(val string) (contract.DeliveryOrder, error) {
 	switch strings.ToLower(val) {
-	case deliveryOrderOrdered:
+	case string(internals.Ordered):
 		return contract.DeliveryOrder_ORDERED, nil
-	case deliveryOrderUnordered:
+	case string(internals.Unordered):
 		return contract.DeliveryOrder_UNORDERED, nil
 	default:
-		return contract.DeliveryOrder_UNORDERED, fmt.Errorf("invalid annotation %s value: %s. Allowed values [ %q | %q ]", deliveryOrderAnnotation, val, deliveryOrderOrdered, deliveryOrderUnordered)
+		return contract.DeliveryOrder_UNORDERED, fmt.Errorf("invalid annotation %s value: %s. Allowed values [ %q | %q ]", deliveryOrderAnnotation, val, internals.Ordered, internals.Unordered)
+	}
+}
+
+func (r Reconciler) reconcileConsumerGroup(ctx context.Context, trigger *eventing.Trigger, statusConditionManager statusConditionManager) error {
+	replicas := int32(1) //todo
+	expectedcg := &internalscg.ConsumerGroup{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: internalscg.ConsumerGroupGroupVersionKind.String(),
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      string(trigger.UID),
+			Namespace: trigger.Namespace,
+		},
+		Spec: internalscg.ConsumerGroupSpec{
+			Template: internalscg.ConsumerTemplateSpec{
+				Spec: internalscg.ConsumerSpec{},
+			},
+			Replicas: &replicas,
+		},
 	}
+
+	cg, err := r.ConsumerGroupLister.ConsumerGroups(trigger.GetNamespace()).Get(string(trigger.UID)) //Get by consumer group name
+	if err != nil && !apierrors.IsNotFound(err) {
+		return statusConditionManager.failedToReconcileConsumerGroup("failed to get consumer group", err)
+	}
+
+	if apierrors.IsNotFound(err) {
+		if _, err := r.InternalsClient.InternalV1alpha1().ConsumerGroups(expectedcg.GetNamespace()).Create(ctx, expectedcg, metav1.CreateOptions{}); err != nil {
+			return fmt.Errorf("failed to create consumer group%s/%s: %w", expectedcg.GetNamespace(), expectedcg.GetName(), err)
+		}
+		fmt.Printf("created consumer group%s/%s", expectedcg.GetNamespace(), expectedcg.GetName())
+		return nil
+	}
+
+	if equality.Semantic.DeepDerivative(cg.Spec, expectedcg.Spec) {
```

Comment:
Due to the way `DeepDerivative` works we want to compare the fields that are set in expected with the current spec.
```suggestion
	if equality.Semantic.DeepDerivative(expectedcg.Spec, cg.Spec) {
```